### PR TITLE
Fetching data from an api

### PR DESCRIPTION
**Pull request by MUFTIATBAKARE**

### CODE REVIEW

File: src/components/Cards.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { React, useState, useEffect } from "react";
+import BackgroundUrls from "./BackgroundUrls";
+
+const Cards = () => {
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState(null);
+  const [error, setError] = useState(null);
```
Comment (tundeojediran) [inline src/components/Cards.jsx:7]: Good job with the multiple states

File: src/components/Cards.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      })
+      .catch((error) => {
+        console.log(error);
+        setError(error);
+        setData(null);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
```
Comment (tundeojediran) [inline src/components/Cards.jsx:37]: Good use of useEffect to handle the data fetching side-effect

File: src/index.css [hunk 3]
```diff
@@ -25,9 +26,11 @@ a:hover {
...
 body {
   margin: 0;
-  display: flex;
-  place-items: center;
-  min-width: 320px;
+  padding: 0;
+  /* display: flex; */
+  box-sizing: border-box;
+  /* place-items: center; */
+  /* min-width: 320px; */
```
Comment (tundeojediran) [inline src/index.css]: remove commented code when no longer needed

File: src/components/Cards.jsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          </g>
+        </svg>
+      </header>
+      {loading && <div>Data is loading. Please wait...</div>}
+      {error && <div>{`There is a problem fetching your data - ${error}`}</div>}
+      <ul>
+        {data &&
+          data.map((movie, index) => {
+            const bgIndex = index % BackgroundUrls.length;
+            const bgUrl = BackgroundUrls[bgIndex];
```
Comment (tundeojediran) [inline src/components/Cards.jsx]: Good job with the background image rendering.

File: src/components/Cards.jsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            </g>
+          </g>
+        </svg>
+      </header>
+      {loading && <div>Data is loading. Please wait...</div>}
+      {error && <div>{`There is a problem fetching your data - ${error}`}</div>}
+      <ul>
+        {data &&
+          data.map((movie, index) => {
+            const bgIndex = index % BackgroundUrls.length;
```
Comment (tundeojediran) [inline src/components/Cards.jsx]: What does this line do?

**Review by tundeojediran — CHANGES_REQUESTED** [review]
Overall. You did an amazing work. I have got some comments you need to address before I approve.

File: src/components/Cards.jsx (hunk 5 above)
Comment (tundeojediran) [reply src/components/Cards.jsx]: `const bgUrl = BackgroundUrls[index];`

File: src/components/Cards.jsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    fetch(`https://swapi.dev/api/films`)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`Status is ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((rawData) => {
+        console.log(rawData);
+        setData(rawData.results);
```
Comment (tundeojediran) [inline src/components/Cards.jsx]: let duplicate = [...rawData.results]

const newResults = duplicate.map((item, index) => item.imageUrl = BackgroundUrls[index])

setData(newResults)

File: src/components/Cards.jsx (hunk 1 above)
Comment (MUFTIATBAKARE) [reply src/components/Cards.jsx:7]: Thank you

File: src/components/Cards.jsx (hunk 5 above)
Comment (MUFTIATBAKARE) [reply src/components/Cards.jsx]: It creates an id for each background image, just a way to escape the error message in the console, when index is used.

File: src/components/Cards.jsx (hunk 4 above)
Comment (MUFTIATBAKARE) [reply src/components/Cards.jsx]: Thank you, I'll implement a better way

File: src/components/Cards.jsx (hunk 6 above)
Comment (MUFTIATBAKARE) [reply src/components/Cards.jsx]: Ok, Thank you

File: src/components/Cards.jsx (hunk 2 above)
Comment (MUFTIATBAKARE) [reply src/components/Cards.jsx:37]: Thank you

File: src/index.css (hunk 3 above)
Comment (MUFTIATBAKARE) [reply src/index.css]: Ok, I will do that.

Comment (MUFTIATBAKARE) [reply src/index.css]: Done!

File: src/components/Cards.jsx (hunk 6 above)
Comment (MUFTIATBAKARE) [reply src/components/Cards.jsx]: Done, though setData(newResults) gave a blank page.
While debugging, I found out newResults contains only the images, while duplicate contains all necessary data. 

Is it ok to merge, now?

**Pull request closed without merging**